Framework/src/TaskRunnerFactory.cxx

```diff
@@ -62,18 +64,21 @@ TaskRunnerConfig TaskRunnerFactory::extractConfig(const CommonSpec& globalConfig
   if (!taskSpec.dataSource.isOneOf(DataSourceType::DataSamplingPolicy, DataSourceType::Direct)) {
     throw std::runtime_error("This data source of the task '" + taskSpec.taskName + "' is not supported.");
   }
-  auto cycleDurationSeconds = taskSpec.cycleDurationSeconds;
-  auto dummyDatabaseUsed = globalConfig.database.count("implementation") > 0 && globalConfig.database.at("implementation") == "Dummy";
-  if (!dummyDatabaseUsed && cycleDurationSeconds < 10) {
-    ILOG(Error, Support) << "Cycle duration is too short (" << cycleDurationSeconds << "), replaced by a duration of 10 seconds." << ENDM;
-    cycleDurationSeconds = 10;
+
+  // cycle duration
+  // Two ways of configuring, incompatible.
+  // 1. simple, old, way: cycleDurationSeconds is the duration in seconds for all cycles
+  // 2. complex, new, way: cycleDurations: a list of tuples specifying different durations to be applied for a certain time
+  if (taskSpec.cycleDurationSeconds > 0 && taskSpec.cycleDurations.size() > 0) {
+    ILOG(Error, Ops) << "Both cycleDurationSeconds and cycleDurations have been defined. Pick one. Sheepishly exiting out." << ENDM;
+    exit(1);
```

Comment:
perhaps you can throw an exception to be consistent with other fatal errors? this way DPL/FairMQ may handle the exit more gracefully.

Reply:
yes